netmiko/nokia/nokia_srl.py

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2014 - 2022 Kirk Byers
+# Copyright (c) 2014 - 2022 Twin Bridges Technology
+# Copyright (c) 2019 - 2020 NOKIA Inc.
+# MIT License - See License file at:
+#   https://github.com/ktbyers/netmiko/blob/develop/LICENSE
+
+import re
+from typing import Any, Optional, Sequence, TextIO, Union
+from netmiko import log
+from netmiko.no_enable import NoEnable
+from netmiko.base_connection import BaseConnection
+
+
+class NokiaSrlSSH(BaseConnection, NoEnable):
+    """
+    Implement methods for interacting with Nokia SRL devices for SSH.
+
+    Not applicable in Nokia SRL:
+        - check_enable_mode()
+        - enable()
+        - exit_enable_mode()
+
+    Overriden methods to adapt Nokia SR OS behavior (changed):
+        - session_preparation()
+        - set_base_prompt()
+        - config_mode()
+        - exit_config_mode()
+        - check_config_mode()
+        - save_config()
+        - commit()
+        - strip_prompt()
+        - enable()
+        - check_enable_mode()
+
+    By default, the SR Linux CLI prompt consists of two lines of text,
+    indicating with an asterisk whether the configuration has been modified or
+    a plus sign whether the configuration has been saved, the current mode and
+    session type, the current CLI context, and the host name of the SR Linux device.
+
+    Examples:
+
+    --{ running }--[ interface ethernet-1/1 subinterface 1 ]--
+    A:ams01#
+
+    --{ * candidate private private-admin }--[ interface ethernet-1/1 subinterface 1 ]--
+    A:ams01#
+
+    --{ + candidate private private-admin }--[ interface ethernet-1/1 subinterface 1 ]--
+    A:ams01#
+
+    This class support the default prompt configuration.
+    """
+
+    def session_preparation(self) -> None:
+        self._test_channel_read(pattern=r"#")
+        self.ansi_escape_codes = True
+        # Bottom toolbar text not required
+        commands = [
+            "environment cli-engine type basic",
+            "environment complete-on-space false",
+        ]
+        for command in commands:
+            self.disable_paging(command=command, cmd_verify=False, pattern=r"#")
```

Comment:
Does it work properly here with `cmd_verify=True`? It is much better to have cmd_verify=True here unless there is some reason it doesn't work.

Reply:
> Does it work properly here with `cmd_verify=True`? It is much better to have cmd_verify=True here unless there is some reason it doesn't work.

You can update with the following piece of code and it works. I tried to update the new branch but I don't have the right.

`commands = [
            r"environment complete-on-space false",
            r"environment cli-engine type basic",
        ]
        for command in commands:
            self.disable_paging(command=command, cmd_verify=True, pattern=r"#")`

Reply:
Sounds good. I will update that now.